subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/PlatformDependencyModifiers.java

```diff
@@ -0,0 +1,150 @@
+import org.gradle.api.artifacts.MinimalExternalModuleDependency;
+import org.gradle.api.artifacts.ModuleDependency;
+import org.gradle.api.artifacts.dsl.DependencyFactory;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.provider.ProviderConvertible;
+
+import javax.annotation.Nullable;
+
+/**
+ * Dependency APIs for using <a href="https://docs.gradle.org/current/userguide/java_platform_plugin.html#java_platform_plugin">Platforms</a> in {@code dependencies} blocks.
+ *
+ * @since 7.6
+ */
+@Incubating
+public interface PlatformDependencyModifiers extends Dependencies {
+
+    /**
+     * Creates an {@link ExternalModuleDependency} for the given dependency notation and modifies it to select the Platform variant of the given module.
+     *
+     * @param dependencyNotation dependency notation
+     * @return the modified dependency
+     * @see DependencyFactory#create(CharSequence) Valid dependency notation for this method
+     */
+    default ExternalModuleDependency platform(CharSequence dependencyNotation) {
+        return platform(getDependencyFactory().create(dependencyNotation));
+    }
+
+    /**
+     * Creates an {@link ExternalModuleDependency} for the given group, name and version and modifies it to select the Platform variant of the given module.
+     *
+     * @param group the group
+     * @param name the name
+     * @param version the version
+     * @return the modified dependency
+     * @see DependencyFactory#create(String, String, String)
+     */
+    default ExternalModuleDependency platform(@Nullable String group, String name, @Nullable String version) {
+        return platform(getDependencyFactory().create(group, name, version));
+    }
+
+    /**
+     * Takes a given {@link ModuleDependency} and modifies it to select the Platform variant of the given module.
+     *
+     * @param dependency the dependency
+     * @return the modified dependency
+     */
+    <D extends ModuleDependency> D platform(D dependency);
+
+    /**
+     * Takes a given {@code Provider} to a {@link MinimalExternalModuleDependency} and modifies the dependency to select the Platform variant of the given module.
+     *
+     * @param providerConvertibleToDependency the provider
+     * @return a provider to the modified dependency
+     */
+    default Provider<? extends MinimalExternalModuleDependency> platform(ProviderConvertible<? extends MinimalExternalModuleDependency> providerConvertibleToDependency) {
+        return providerConvertibleToDependency.asProvider().map(this::platform);
+    }
+
+    /**
+     * Takes a given {@code Provider} to a {@link ExternalModuleDependency} and modifies the dependency to select the Platform variant of the given module.
+     *
+     * @param providerToDependency the provider
+     * @return a provider to the modified dependency
+     */
+    default Provider<? extends ExternalModuleDependency> platform(Provider<? extends ExternalModuleDependency> providerToDependency) {
```

Comment:
To match `platform(D)`:
```suggestion
    default <D extends ModuleDependency> Provider<D> platform(Provider<D> providerToDependency) {
```